src/heretic/r_bmaps.c

```diff
@@ -0,0 +1,470 @@
+//
+// Copyright(C) 1993-1996 Id Software, Inc.
+// Copyright(C) 2005-2014 Simon Howard
+// Copyright(C) 2013-2017 Brad Harding
+// Copyright(C) 2017-2022 Fabian Greffrath
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//	Brightmaps for walls, floors, sprites and weapon states
+//	Adapted from doomretro/src/r_data.c:97-209
+//
+
+#include "crispy.h"
+#include "doomdef.h"
+#include "r_bmaps.h"
+
+// [crispy] brightmap data
+
+static byte nobrightmap[256] = {0};
+
+/*
+static byte fullbright[256] =
+{
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+};
+*/
+
+static byte surfaces[256] =
```

Comment:
Should these arrays also be `const`?

Reply:
Hm, trying to set them as `static const byte surfaces[256] =` gives me compiler warning:
```
r_bmaps.c:183:18: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  183 |     {"DOOREXIT", surfaces},
      |                  ^~~~~~~~
```

Reply:
I think that means that also `colormap` should be `const` in `fullbright_t`.

Reply:
Figured it out. To make it working, literally everything b-map related should be converted to `const` then, i.e. `R_BrightmapFor*` bytes and `texturebrightmap` itself. But is it really necessary?

Reply:
Not necessary but very useful for the next person who has to tinker with it. With `const` they can see what data isn't being changed in some random place. Also makes it slightly more efficient for memory usage since the kernel might have more read-only pages to evict.